File: k.yyup.com/auto-fix-pages.js
#!/usr/bin/env node

/**
 * 自动化页面修复脚本
 * 使用Claude Code SDK批量修复前端页面问题
 * 支持断线续传和进度保存
 */

const fs = require('fs');
const path = require('path');
const { spawn } = require('child_process');

// 配置
const CONFIG = {
  // Claude Code 命令
  claudeCodeCommand: 'claude-code',
  
  // 项目路径
  projectPath: process.cwd(),
  clientPath: path.join(process.cwd(), 'client'),
  
  // 进度文件
  progressFile: path.join(process.cwd(), '.auto-fix-progress.json'),
  
  // 日志文件
  logFile: path.join(process.cwd(), 'auto-fix.log'),
  
  // 并发数量
  concurrency: 1,
  
  // 重试次数
  maxRetries: 3,
  
  // 延迟时间（毫秒）
  delay: 2000
};

// 需要修复的页面列表
const PAGES_TO_FIX = [
  // 用户管理页面
  { path: 'client/src/views/system/User.vue', category: 'system', priority: 1 },
  { path: 'client/src/views/system/Role.vue', category: 'system', priority: 1 },
  { path: 'client/src/views/system/Permission.vue', category: 'system', priority: 1 },
  
  // 教育管理页面
  { path: 'client/src/views/student/Student.vue', category: 'education', priority: 2 },
  { path: 'client/src/views/student/StudentDetail.vue', category: 'education', priority: 2 },
  { path: 'client/src/views/teacher/Teacher.vue', category: 'education', priority: 2 },
  { path: 'client/src/views/teacher/TeacherDetail.vue', category: 'education', priority: 2 },
  { path: 'client/src/views/teacher/TeacherEdit.vue', category: 'education', priority: 2 },
  { path: 'client/src/views/parent/Parent.vue', category: 'education', priority: 2 },
  { path: 'client/src/views/parent/ParentDetail.vue', category: 'education', priority: 2 },
  { path: 'client/src/views/parent/ParentChildren.vue', category: 'education', priority: 2 },
  { path: 'client/src/views/class/Class.vue', category: 'education', priority: 2 },
  { path: 'client/src/views/class/ClassDetail.vue', category: 'education', priority: 2 },
  
  // 招生管理页面
  { path: 'client/src/views/enrollment/EnrollmentOverview.vue', category: 'enrollment', priority: 3 },
  { path: 'client/src/views/enrollment/EnrollmentPlan.vue', category: 'enrollment', priority: 3 },
  { path: 'client/src/views/enrollment/EnrollmentPlanDetail.vue', category: 'enrollment', priority: 3 },
  { path: 'client/src/views/enrollment/EnrollmentPlanEdit.vue', category: 'enrollment', priority: 3 },
  { path: 'client/src/views/enrollment/EnrollmentPlanStatistics.vue', category: 'enrollment', priority: 3 },
  
  // 活动管理页面
  { path: 'client/src/views/activity/Activity.vue', category: 'activity', priority: 4 },
  { path: 'client/src/views/activity/ActivityCreate.vue', category: 'activity', priority: 4 },
  { path: 'client/src/views/activity/ActivityDetail.vue', category: 'activity', priority: 4 },
  
  // AI系统页面
  { path: 'client/src/views/ai/AIChat.vue', category: 'ai', priority: 5 },
  { path: 'client/src/views/ai/AIAssistant.vue', category: 'ai', priority: 5 },
  { path: 'client/src/views/ai/AIMemory.vue', category: 'ai', priority: 5 },
  { path: 'client/src/views/ai/AIModel.vue', category: 'ai', priority: 5 },
  
  // 系统管理页面
  { path: 'client/src/views/system/SystemSettings.vue', category: 'system', priority: 6 },
  { path: 'client/src/views/system/SystemLogs.vue', category: 'system', priority: 6 },
  { path: 'client/src/views/system/SystemBackup.vue', category: 'system', priority: 6 },
  { path: 'client/src/views/system/AIModelConfig.vue', category: 'system', priority: 6 },
  { path: 'client/src/views/system/MessageTemplate.vue', category: 'system', priority: 6 },
  
  // 园长功能页面
  { path: 'client/src/views/principal/PrincipalDashboard.vue', category: 'principal', priority: 7 },
  { path: 'client/src/views/principal/PrincipalPerformance.vue', category: 'principal', priority: 7 },
  { path: 'client/src/views/principal/PrincipalCustomerPool.vue', category: 'principal', priority: 7 },
  
  // 业务扩展页面
  { path: 'client/src/views/business/Statistics.vue', category: 'business', priority: 8 },
  { path: 'client/src/views/business/Customer.vue', category: 'business', priority: 8 },
  { path: 'client/src/views/business/Advertisement.vue', category: 'business', priority: 8 },
  { path: 'client/src/views/business/Marketing.vue', category: 'business', priority: 8 },
  { path: 'client/src/views/business/Application.vue', category: 'business', priority: 8 },
  { path: 'client/src/views/business/Chat.vue', category: 'business', priority: 8 }
];

// 修复提示词模板
const FIX_PROMPT_TEMPLATE = `
你是一个Vue.js前端开发专家，需要修复幼儿园招生管理系统的页面。

## 当前任务
修复页面：{PAGE_PATH}
页面分类：{CATEGORY}
优先级：{PRIORITY}

## 系统背景
这是一个基于Vue 3 + TypeScript + Element Plus的幼儿园招生管理系统，包含以下功能模块：
- 用户管理（用户、角色、权限）
- 教育管理（学生、教师、家长、班级）
- 招生管理（招生计划、统计）
- 活动管理（活动列表、创建、详情）
- AI系统（对话、助手、记忆、模型）
- 系统管理（设置、日志、备份）
- 园长功能（仪表板、绩效、客户池）
- 业务扩展（统计、客户、广告、营销）

## 技术栈
- Vue 3 Composition API
- TypeScript
- Element Plus UI组件库
- Pinia状态管理
- Vue Router路由
- Axios HTTP客户端

## 修复要求

### 1. 页面结构规范
- 使用Vue 3 Composition API语法
- 正确的TypeScript类型定义
- 合理的组件结构和布局

### 2. UI组件使用
- 使用Element Plus组件库
- 遵循设计系统规范
- 响应式布局设计

### 3. 数据管理
- 使用Pinia进行状态管理
- 正确的API调用和错误处理
- 数据验证和表单验证

### 4. 用户体验
- 加载状态和错误提示
- 操作反馈和确认
- 无障碍访问支持

### 5. 代码质量
- 清晰的代码结构
- 适当的注释说明
- 性能优化考虑

## 具体修复指导

根据页面分类提供具体指导：

### 系统管理页面 (system)
- 实现CRUD操作（增删改查）
- 权限控制和角色验证
- 数据表格和分页
- 搜索和筛选功能

### 教育管理页面 (education)
- 学生/教师/家长信息管理
- 关联关系处理
- 详情页面和编辑功能
- 数据导入导出

### 招生管理页面 (enrollment)
- 招生计划管理
- 统计图表展示
- 报名流程处理
- 数据分析功能

### 活动管理页面 (activity)
- 活动列表和详情
- 活动创建和编辑
- 参与者管理
- 活动状态跟踪

### AI系统页面 (ai)
- 对话界面设计
- AI助手功能
- 记忆管理
- 模型配置

### 园长功能页面 (principal)
- 数据仪表板
- 绩效分析
- 客户池管理
- 决策支持

### 业务扩展页面 (business)
- 统计分析图表
- 客户关系管理
- 营销活动管理
- 应用集成

## 输出要求
1. 修复后的完整Vue文件代码
2. 简要说明修复的问题和改进点
3. 如果需要新增依赖或配置，请说明

请开始修复页面：{PAGE_PATH}
`;

// 工具函数
class AutoFixManager {
  constructor() {
    this.progress = this.loadProgress();
    this.setupLogging();
  }

  // 加载进度
  loadProgress() {
    try {
      if (fs.existsSync(CONFIG.progressFile)) {
        return JSON.parse(fs.readFileSync(CONFIG.progressFile, 'utf8'));
      }
    } catch (error) {
      this.log('警告：无法加载进度文件，将从头开始');
    }
    return {
      completed: [],
      failed: [],
      currentIndex: 0,
      startTime: new Date().toISOString()
    };
  }

  // 保存进度
  saveProgress() {
    try {
      fs.writeFileSync(CONFIG.progressFile, JSON.stringify(this.progress, null, 2));
    } catch (error) {
      this.log(`错误：无法保存进度文件 - ${error.message}`);
    }
  }

  // 设置日志
  setupLogging() {
    this.logStream = fs.createWriteStream(CONFIG.logFile, { flags: 'a' });
  }

  // 记录日志
  log(message) {
    const timestamp = new Date().toISOString();
    const logMessage = `[${timestamp}] ${message}\n`;
    console.log(message);
    if (this.logStream) {
      this.logStream.write(logMessage);
    }
  }

  // 生成修复提示词
  generatePrompt(page) {
    return FIX_PROMPT_TEMPLATE
      .replace(/{PAGE_PATH}/g, page.path)
      .replace(/{CATEGORY}/g, page.category)
      .replace(/{PRIORITY}/g, page.priority);
  }

  // 执行Claude Code命令
  async executeClaudeCode(page, prompt) {
    return new Promise((resolve, reject) => {
      const args = [
        'fix',
        '--file', page.path,
        '--prompt', prompt,
        '--auto-apply'
      ];

      this.log(`执行命令: ${CONFIG.claudeCodeCommand} ${args.join(' ')}`);

      const process = spawn(CONFIG.claudeCodeCommand, args, {
        cwd: CONFIG.projectPath,
        stdio: ['pipe', 'pipe', 'pipe']
      });

      let stdout = '';
      let stderr = '';

      process.stdout.on('data', (data) => {
        stdout += data.toString();
      });

      process.stderr.on('data', (data) => {
        stderr += data.toString();
      });

      process.on('close', (code) => {
        if (code === 0) {
          resolve({ success: true, output: stdout });
        } else {
          reject(new Error(`命令执行失败 (退出码: ${code})\n${stderr}`));
        }
      });

      process.on('error', (error) => {
        reject(new Error(`无法启动进程: ${error.message}`));
      });
    });
  }

  // 修复单个页面
  async fixPage(page, retryCount = 0) {
    try {
      this.log(`开始修复页面: ${page.path} (分类: ${page.category}, 优先级: ${page.priority})`);

      // 检查文件是否存在
      const fullPath = path.join(CONFIG.projectPath, page.path);
      if (!fs.existsSync(fullPath)) {
        this.log(`警告：文件不存在，跳过 - ${page.path}`);
        return { success: false, reason: 'file_not_found' };
      }

      // 生成提示词
      const prompt = this.generatePrompt(page);

      // 执行修复
      const result = await this.executeClaudeCode(page, prompt);

      this.log(`✅ 页面修复成功: ${page.path}`);
      return { success: true, output: result.output };

    } catch (error) {
      this.log(`❌ 页面修复失败: ${page.path} - ${error.message}`);

      if (retryCount < CONFIG.maxRetries) {
        this.log(`重试 ${retryCount + 1}/${CONFIG.maxRetries}: ${page.path}`);
        await this.delay(CONFIG.delay * (retryCount + 1));
        return this.fixPage(page, retryCount + 1);
      }

      return { success: false, error: error.message };
    }
  }

  // 延迟函数
  delay(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
  }

  // 主执行函数
  async run() {
    this.log('🚀 开始自动修复页面');
    this.log(`总页面数: ${PAGES_TO_FIX.length}`);
    this.log(`已完成: ${this.progress.completed.length}`);
    this.log(`已失败: ${this.progress.failed.length}`);

    // 按优先级排序
    const sortedPages = PAGES_TO_FIX.sort((a, b) => a.priority - b.priority);

    // 从上次中断的地方继续
    for (let i = this.progress.currentIndex; i < sortedPages.length; i++) {
      const page = sortedPages[i];

      // 跳过已完成的页面
      if (this.progress.completed.includes(page.path)) {
        this.log(`跳过已完成的页面: ${page.path}`);
        continue;
      }

      // 更新当前索引
      this.progress.currentIndex = i;
      this.saveProgress();

      // 修复页面
      const result = await this.fixPage(page);

      if (result.success) {
        this.progress.completed.push(page.path);
      } else {
        this.progress.failed.push({
          path: page.path,
          reason: result.reason || result.error,
          timestamp: new Date().toISOString()
        });
      }

      // 保存进度
      this.saveProgress();

      // 延迟避免过于频繁的调用
      if (i < sortedPages.length - 1) {
        await this.delay(CONFIG.delay);
      }
    }

    // 完成总结
    this.log('🎉 自动修复完成！');
    this.log(`✅ 成功: ${this.progress.completed.length}`);
    this.log(`❌ 失败: ${this.progress.failed.length}`);

    if (this.progress.failed.length > 0) {
      this.log('\n失败的页面:');
      this.progress.failed.forEach(failed => {
        this.log(`  - ${failed.path}: ${failed.reason}`);
      });
    }

    // 清理
    if (this.logStream) {
      this.logStream.end();
    }
  }
}

// 命令行参数处理
function parseArgs() {
  const args = process.argv.slice(2);
  const options = {};

  for (let i = 0; i < args.length; i++) {
    switch (args[i]) {
      case '--reset':
        options.reset = true;
        break;
      case '--category':
        options.category = args[++i];
        break;
      case '--priority':
        options.priority = parseInt(args[++i]);
        break;
      case '--help':
        options.help = true;
        break;
    }
  }

  return options;
}

// 显示帮助信息
function showHelp() {
  console.log(`
自动化页面修复脚本

用法:
  node auto-fix-pages.js [选项]

选项:
  --reset              重置进度，从头开始
  --category <类型>    只修复指定分类的页面
  --priority <数字>    只修复指定优先级的页面
  --help              显示此帮助信息

示例:
  node auto-fix-pages.js                    # 修复所有页面
  node auto-fix-pages.js --reset            # 重置进度重新开始
  node auto-fix-pages.js --category system  # 只修复系统管理页面
  node auto-fix-pages.js --priority 1       # 只修复优先级1的页面
`);
}

// 主函数
async function main() {
  const options = parseArgs();

  if (options.help) {
    showHelp();
    return;
  }

  // 重置进度
  if (options.reset && fs.existsSync(CONFIG.progressFile)) {
    fs.unlinkSync(CONFIG.progressFile);
    console.log('✅ 进度已重置');
  }

  // 过滤页面
  let pagesToFix = PAGES_TO_FIX;
  if (options.category) {
    pagesToFix = pagesToFix.filter(page => page.category === options.category);
  }
  if (options.priority) {
    pagesToFix = pagesToFix.filter(page => page.priority === options.priority);
  }

  // 更新页面列表
  PAGES_TO_FIX.length = 0;
  PAGES_TO_FIX.push(...pagesToFix);

  // 创建管理器并运行
  const manager = new AutoFixManager();
  await manager.run();
}

// 错误处理
process.on('uncaughtException', (error) => {
  console.error('未捕获的异常:', error);
  process.exit(1);
});

process.on('unhandledRejection', (reason, promise) => {
  console.error('未处理的Promise拒绝:', reason);
  process.exit(1);
});

// 运行
if (require.main === module) {
  main().catch(console.error);
}

module.exports = { AutoFixManager, CONFIG, PAGES_TO_FIX };

File: k.yyup.com/auto_fix_pages.py
#!/usr/bin/env python3
"""
自动化页面修复脚本
使用Claude Code Python SDK批量修复前端页面问题
支持断线续传和进度保存
"""

import asyncio
import json
import logging
import os
import sys
from datetime import datetime
from pathlib import Path
from typing import Dict, List, Optional, Any
import argparse

# 安装依赖: pip install claude-code-sdk anyio
try:
    from claude_code_sdk import query, ClaudeCodeOptions, Message
    import anyio
except ImportError:
    print("❌ 请先安装依赖: pip install claude-code-sdk anyio")
    sys.exit(1)

# 配置
class Config:
    PROJECT_PATH = Path.cwd()
    CLIENT_PATH = PROJECT_PATH / "client"
    PROGRESS_FILE = PROJECT_PATH / ".auto_fix_progress.json"
    LOG_FILE = PROJECT_PATH / "auto_fix.log"
    MAX_RETRIES = 3
    DELAY_SECONDS = 2
    MAX_TURNS = 5

# 需要修复的页面列表 (基于实际存在的文件)
PAGES_TO_FIX = [
    # AI系统页面 (优先级1 - 实际存在的文件)
    {"path": "client/src/views/ai/MemoryManagement.vue", "category": "ai", "priority": 1, "description": "AI记忆管理页面"},

    # 活动管理页面 (优先级2 - 实际存在的文件)
    {"path": "client/src/views/principal/activity/index.vue", "category": "activity", "priority": 2, "description": "活动管理页面"},

    # 登录页面 (优先级3 - 实际存在的文件)
    {"path": "client/src/views/Login/index.vue", "category": "system", "priority": 3, "description": "用户登录页面"},
]

# 精准修复提示词模板
FIX_PROMPT_TEMPLATE = """
你是一个Vue.js前端开发专家，专门修复幼儿园招生管理系统的页面问题。

## 当前任务
修复页面：{page_path}
页面分类：{category}
页面描述：{description}

## 系统背景
这是一个基于Vue 3 + TypeScript + Element Plus的幼儿园招生管理系统。
后端API地址：http://localhost:3000/api
前端运行地址：http://localhost:5173

## 技术栈要求
- Vue 3 Composition API (使用 <script setup lang="ts">)
- TypeScript (严格类型检查)
- Element Plus UI组件库
- Pinia状态管理 (stores在 @/stores/ 目录)
- Vue Router (路由配置在 @/router/)
- Axios HTTP客户端 (API调用在 @/api/ 目录)

## 核心修复目标

### 1. 移除硬编码数据
❌ 错误示例：
```javascript
const stats = reactive({{
  totalCount: 128,  // 硬编码数据
  averageImportance: 0.68
}});
```

✅ 正确示例：
```javascript
const stats = reactive({{
  totalCount: 0,
  averageImportance: 0,
  loading: false,
  error: null
}});

// 从API获取真实数据
const fetchStats = async () => {{
  try {{
    stats.loading = true;
    const response = await api.getMemoryStats();
    Object.assign(stats, response.data);
  }} catch (error) {{
    stats.error = error.message;
    ElMessage.error('获取统计数据失败');
  }} finally {{
    stats.loading = false;
  }}
}};
```

### 2. 实现完整的API集成
必须包含：
- API调用函数 (使用 @/api/ 目录下的模块)
- 加载状态管理 (loading: boolean)
- 错误处理 (try-catch + ElMessage)
- 数据验证 (检查响应数据格式)

### 3. 完善TypeScript类型定义
```typescript
// 定义接口类型
interface DataStats {{
  totalCount: number;
  averageValue: number;
  distribution: Record<string, number>;
}}

// 定义组件状态类型
interface ComponentState {{
  loading: boolean;
  error: string | null;
  data: DataStats | null;
}}
```

### 4. 添加用户体验优化
- 加载骨架屏或loading状态
- 空数据状态展示
- 错误状态处理和重试机制
- 操作成功/失败的用户反馈

### 5. 遵循组件结构规范
```vue
<template>
  <!-- 使用v-loading指令 -->
  <div v-loading="loading" class="page-container">
    <!-- 错误状态 -->
    <el-alert v-if="error" type="error" :title="error" show-icon />

    <!-- 空数据状态 -->
    <el-empty v-else-if="!loading && !data" description="暂无数据" />

    <!-- 正常内容 -->
    <div v-else>
      <!-- 页面内容 -->
    </div>
  </div>
</template>

<script setup lang="ts">
import {{ ref, reactive, onMounted }} from 'vue';
import {{ ElMessage }} from 'element-plus';

// 状态管理
const state = reactive({{
  loading: false,
  error: null,
  data: null
}});

// API调用
const fetchData = async () => {{
  // 实现API调用逻辑
}};

// 生命周期
onMounted(() => {{
  fetchData();
}});
</script>

<style scoped>
.page-container {{
  padding: 20px;
}}
</style>
```

## 具体修复指导

### AI系统页面 (ai)
- 集成AI相关API (对话、记忆、模型管理)
- 实现实时数据更新
- 添加AI交互状态管理
- 优化图表和数据可视化

### 活动管理页面 (activity)
- 实现CRUD操作API集成
- 添加活动状态管理
- 实现搜索和筛选功能
- 添加活动时间管理

### 园长功能页面 (principal)
- 集成权限验证
- 实现数据仪表板
- 添加统计图表
- 实现导出功能

### 系统管理页面 (system)
- 实现用户权限管理
- 添加系统配置API
- 实现日志查看功能
- 添加数据备份恢复

## 输出要求
1. 修复后的完整Vue文件代码
2. 确保所有硬编码数据都替换为API调用
3. 添加完整的TypeScript类型定义
4. 实现加载状态和错误处理
5. 优化用户体验和界面交互

## 注意事项
- 保持现有的UI设计风格
- 确保响应式布局正常工作
- 添加适当的注释说明
- 遵循Vue 3最佳实践
- 确保代码可维护性和可读性

请开始修复页面：{page_path}
"""

class AutoFixManager:
    def __init__(self):
        self.setup_logging()
        self.progress = self.load_progress()
        
    def setup_logging(self):
        """设置日志"""
        logging.basicConfig(
            level=logging.INFO,
            format='%(asctime)s - %(levelname)s - %(message)s',
            handlers=[
                logging.FileHandler(Config.LOG_FILE),
                logging.StreamHandler()
            ]
        )
        self.logger = logging.getLogger(__name__)
        
    def load_progress(self) -> Dict[str, Any]:
        """加载进度"""
        try:
            if Config.PROGRESS_FILE.exists():
                with open(Config.PROGRESS_FILE, 'r', encoding='utf-8') as f:
                    return json.load(f)
        except Exception as e:
            self.logger.warning(f"无法加载进度文件: {e}")
            
        return {
            "completed": [],
            "failed": [],
            "current_index": 0,
            "start_time": datetime.now().isoformat()
        }
        
    def save_progress(self):
        """保存进度"""
        try:
            with open(Config.PROGRESS_FILE, 'w', encoding='utf-8') as f:
                json.dump(self.progress, f, indent=2, ensure_ascii=False)
        except Exception as e:
            self.logger.error(f"无法保存进度文件: {e}")
            
    def generate_prompt(self, page: Dict[str, Any]) -> str:
        """生成修复提示词"""
        return FIX_PROMPT_TEMPLATE.format(
            page_path=page["path"],
            category=page["category"],
            priority=page["priority"],
            description=page["description"]
        )
        
    async def fix_page(self, page: Dict[str, Any], retry_count: int = 0) -> Dict[str, Any]:
        """修复单个页面"""
        try:
            self.logger.info(f"开始修复页面: {page['path']} (分类: {page['category']}, 优先级: {page['priority']})")

            # 检查文件是否存在
            full_path = Config.PROJECT_PATH / page["path"]
            if not full_path.exists():
                self.logger.warning(f"文件不存在，跳过: {page['path']}")
                return {"success": False, "reason": "file_not_found"}

            # 生成提示词
            prompt = self.generate_prompt(page)

            # 配置Claude Code选项
            options = ClaudeCodeOptions(
                max_turns=Config.MAX_TURNS,
                cwd=Config.PROJECT_PATH,
                allowed_tools=["Read", "Write", "Bash"],
                permission_mode="acceptEdits"
            )

            # 执行修复
            messages: List[Message] = []
            message_count = 0
            async for message in query(prompt=prompt, options=options):
                messages.append(message)
                message_count += 1
                self.logger.info(f"📨 收到消息 {message_count}: {type(message).__name__}")

            # 处理结果
            if messages:
                self.logger.info(f"✅ 页面修复完成: {page['path']} (收到 {len(messages)} 条消息)")
                return {
                    "success": True,
                    "messages": len(messages),
                    "details": "修复完成"
                }
            else:
                self.logger.warning(f"⚠️ 页面修复无响应: {page['path']}")
                return {"success": False, "reason": "no_response"}

        except Exception as e:
            self.logger.error(f"❌ 页面修复失败: {page['path']} - {str(e)}")

            if retry_count < Config.MAX_RETRIES:
                self.logger.info(f"重试 {retry_count + 1}/{Config.MAX_RETRIES}: {page['path']}")
                await asyncio.sleep(Config.DELAY_SECONDS * (retry_count + 1))
                return await self.fix_page(page, retry_count + 1)

            return {"success": False, "error": str(e)}
            
    async def run(self, category_filter: Optional[str] = None, priority_filter: Optional[int] = None):
        """主执行函数"""
        self.logger.info("🚀 开始自动修复页面")
        
        # 过滤页面
        pages_to_fix = PAGES_TO_FIX
        if category_filter:
            pages_to_fix = [p for p in pages_to_fix if p["category"] == category_filter]
        if priority_filter:
            pages_to_fix = [p for p in pages_to_fix if p["priority"] == priority_filter]
            
        # 按优先级排序
        pages_to_fix.sort(key=lambda x: x["priority"])
        
        self.logger.info(f"总页面数: {len(pages_to_fix)}")
        self.logger.info(f"已完成: {len(self.progress['completed'])}")
        self.logger.info(f"已失败: {len(self.progress['failed'])}")
        
        # 从上次中断的地方继续
        for i, page in enumerate(pages_to_fix[self.progress["current_index"]:], self.progress["current_index"]):
            # 跳过已完成的页面
            if page["path"] in self.progress["completed"]:
                self.logger.info(f"跳过已完成的页面: {page['path']}")
                continue
                
            # 更新当前索引
            self.progress["current_index"] = i
            self.save_progress()
            
            # 修复页面
            result = await self.fix_page(page)
            
            if result["success"]:
                self.progress["completed"].append(page["path"])
                self.logger.info(f"✅ 成功修复: {page['path']}")
            else:
                self.progress["failed"].append({
                    "path": page["path"],
                    "reason": result.get("reason", result.get("error", "unknown")),
                    "timestamp": datetime.now().isoformat()
                })
                self.logger.error(f"❌ 修复失败: {page['path']}")
                
            # 保存进度
            self.save_progress()
            
            # 延迟避免过于频繁的调用
            if i < len(pages_to_fix) - 1:
                await asyncio.sleep(Config.DELAY_SECONDS)
                
        # 完成总结
        self.logger.info("🎉 自动修复完成！")
        self.logger.info(f"✅ 成功: {len(self.progress['completed'])}")
        self.logger.info(f"❌ 失败: {len(self.progress['failed'])}")
        
        if self.progress["failed"]:
            self.logger.info("\n失败的页面:")
            for failed in self.progress["failed"]:
                self.logger.info(f"  - {failed['path']}: {failed['reason']}")

def main():
    """主函数"""
    parser = argparse.ArgumentParser(description="自动化页面修复脚本")
    parser.add_argument("--reset", action="store_true", help="重置进度，从头开始")
    parser.add_argument("--category", type=str, help="只修复指定分类的页面")
    parser.add_argument("--priority", type=int, help="只修复指定优先级的页面")
    parser.add_argument("--list", action="store_true", help="列出所有页面")
    
    args = parser.parse_args()
    
    if args.list:
        print("📋 所有页面列表:")
        for page in sorted(PAGES_TO_FIX, key=lambda x: (x["priority"], x["category"])):
            print(f"  优先级{page['priority']} | {page['category']} | {page['path']} | {page['description']}")
        return
        
    # 重置进度
    if args.reset and Config.PROGRESS_FILE.exists():
        Config.PROGRESS_FILE.unlink()
        print("✅ 进度已重置")
        
    # 创建管理器并运行
    manager = AutoFixManager()
    
    try:
        anyio.run(manager.run, args.category, args.priority)
    except KeyboardInterrupt:
        print("\n⚠️ 用户中断，进度已保存")
    except Exception as e:
        print(f"❌ 执行失败: {e}")
        sys.exit(1)

if __name__ == "__main__":
    main()
